frontend/talentsearch/src/js/components/workPreferencesForm/WorkPreferencesForm.tsx

```diff
@@ -1,26 +1,96 @@
-import { BasicForm, Checklist, RadioGroup } from "@common/components/form";
+import { commonMessages, errorMessages } from "@common/messages";
+import { Checklist, RadioGroup } from "@common/components/form";
 import { getOperationalRequirementCandidateDescription } from "@common/constants/localizedConstants";
 import { enumToOptions } from "@common/helpers/formUtils";
-import { OperationalRequirement } from "../../api/generated";
-import ProfileFormWrapper from "../applicantProfile/ProfileFormWrapper";
+import { navigate } from "@common/helpers/router";
+import { toast } from "react-toastify";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import { useApplicantProfileRoutes } from "../../applicantProfileRoutes";
 import ProfileFormFooter from "../applicantProfile/ProfileFormFooter";
+import ProfileFormWrapper from "../applicantProfile/ProfileFormWrapper";
+import {
+  GetWorkPreferencesQuery,
+  OperationalRequirement,
+  UpdateUserAsUserInput,
+  UpdateWorkPreferencesMutation,
+  useGetWorkPreferencesQuery,
+  useUpdateWorkPreferencesMutation,
+} from "../../api/generated";
 
-export type FormValues = {
-  requiredWorkPreferences: string;
-  optionalWorkPreferences: string[];
+export type FormValues = Pick<
+  UpdateUserAsUserInput,
+  "acceptedOperationalRequirements"
+> & {
+  wouldAcceptTemporary: string;
 };
+export interface WorkPreferencesFormProps {
+  initialData: GetWorkPreferencesQuery | undefined;
+  handleWorkPreferences: (
+    id: string,
+    data: UpdateUserAsUserInput,
+  ) => Promise<UpdateWorkPreferencesMutation["updateUserAsUser"]>;
+}
 
-export const WorkPreferencesForm: React.FunctionComponent<{
-  handleSubmit: (data: FormValues) => Promise<void>;
-}> = ({ handleSubmit }) => {
+export const WorkPreferencesForm: React.FC<WorkPreferencesFormProps> = ({
+  initialData,
+  handleWorkPreferences,
+}) => {
   const intl = useIntl();
-
   function bold(msg: string) {
     return <span data-h2-font-weight="b(700)">{msg}</span>;
   }
+  const OperationalRequirementsSortOrder = [
+    OperationalRequirement.OvertimeShortNotice,
+    OperationalRequirement.OvertimeScheduled,
+    OperationalRequirement.ShiftWork,
+    OperationalRequirement.OnCall,
+    OperationalRequirement.Travel,
+    OperationalRequirement.TransportEquipment,
+    OperationalRequirement.DriversLicense,
+  ];
+  const dataToFormValues = (
+    data?: GetWorkPreferencesQuery | undefined,
+  ): FormValues => {
+    const boolToString = (boolVal: boolean | null | undefined): string => {
+      return boolVal ? "true" : "false";
+    };
+
+    return {
+      wouldAcceptTemporary: boolToString(data?.me?.wouldAcceptTemporary),
+      acceptedOperationalRequirements:
+        data?.me?.acceptedOperationalRequirements,
+    };
+  };
+  const formValuesToSubmitData = (
+    values: FormValues,
+  ): UpdateUserAsUserInput => {
+    const StringtoBool = (stringVal: string): boolean | null | undefined => {
```

Comment:
Super nitpicking here but can we change the case on this so it starts with a lower case "s" i.e `stringToBool`?

Reply:
Sure! Thank you!